File: server/models/transaction.py
"""SQLAlchemy model for the transactions table."""
from sqlalchemy import Column, String, Integer, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from server.models.agent import Base


class Transaction(Base):
    """Represents a payment transaction executed by an agent."""

    __tablename__ = "transactions"

    id = Column(String, primary_key=True)
    agent_did = Column(String, ForeignKey("agents.did"), nullable=False)
    amount = Column(Integer, nullable=False)
    currency = Column(String, default='jpy')
    description = Column(Text, nullable=True)
    status = Column(String, nullable=False)
    stripe_payment_intent_id = Column(String, nullable=True)
    idempotency_key = Column(String, unique=True, nullable=True)
    audit_hash = Column(String, nullable=True)
    created_at = Column(DateTime, server_default=func.now())

File: server-rust/src/models/transaction.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

#[derive(Debug, sqlx::FromRow)]
pub struct Transaction {
    pub id: String,
    pub agent_did: String,
    pub amount: i64,
    pub currency: String,
    pub description: Option<String>,
    pub status: String,
    pub payment_provider: Option<String>,
    pub provider_payment_id: Option<String>,
    pub idempotency_key: Option<String>,
    pub approval_id: Option<Uuid>,
    pub audit_hash: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
